code/modules/hydroponics/seeds.dm

```diff
@@ -54,181 +54,241 @@
 	update_seed()
 */
 /obj/item/seeds/poppyseed
+	name = "packet of poppy seeds"
```

Comment:
The name already gets set by base seed code, if that isnt working properly this isn't the correct fix

Reply:
It gets set only when the seed is dispensed with the /New() proc, the names of the seeds in the vendor are still "packet of seeds". This may be a bit hacky but I have no better idea on how to solve it.

Reply:
Looks like there's a global list of seed_types you can grab the name from for the vendor